namespaces/mson-namespace.md

```diff
@@ -0,0 +1,503 @@
+
+# Terminology
+
+This specification uses terminology from [MSON Specification][], in addition, bellow you can find the explanation of terms defined in this namespace.
+
+## Expanded Element
+
+MSON is built around the idea of defining recursive data structures. To provide abstraction, for convenience reasons and to not repeat ourselves these structures can be named (using an _identifier_) and reused. In MSON, the reusable data structures are called _Named Types_.
+
+Often, before an MSON DOM can be processed referenced _Named Types_ have to be resolved. Resolving references to _Named Types_ is tedious and error prone. As such parser can resolve references and produce a complete MSON DOM, that is a DOM that does not include unresolved references to other data structures. This is referred to as _reference expansion_ or simply _expansion_.
+
+In other words, an expanded element is such an element that does not contain _Identifier_ (defined bellow) to any other elements but those defined in JSON or MSON namespaces.
+
+The expanded DOM MUST, however, keeps the track of what data structure was expanded and what from where.
+
+## Base Element
+
+In MSON, every data structure is a sub-type of another data structure, and, in transition, it is directly or indirectly derived from one of the MSON _Base Types_. This is expressed as an inheritance of elements in MSON DOM. Where the predecessor of an element is referred to as element's _Base Element_.
+
+Note not every MSON _Base Type_ is presented in JSON namespace primitive types and vice versa, see the table bellow:
+
+### JSON Namespace vs. MSON built-in types
+
+| JSON primitive |   JSON Namespace   | MSON Base Type |   MSON Namespace   |
+|:--------------:|:------------------:|:--------------:|:------------------:|
+|     boolean    |  JSON:Boolean Type |     boolean    |  MSON:Boolean Type |
+|     string     |  JSON:String Type  |     string     |  MSON:String Type  |
+|     number     |  JSON:Number Type  |     number     |  MSON:Number Type  |
+|      array     |   JSON:Array Type  |      array     |   MSON:Array Type  |
+|        -       |          -         |      enum      |   MSON:Enum Type   |
+|     object     |  JSON:Object Type  |     object     |  MSON:Object Type  |
+|      null      |   JSON:Null Type   |        -       |          -         |
+|        -       | JSON:Property Type |        -       | MSON:Property Type |
+
+
+# General Elements Definition
+
+General elements defined inside MSON namespaces but possibly reusable in another domain.
+
+## Identifier (Enum)
+
+Identifier of an element in the MSON namespace. The identifier MUST be either directly a `string` value or an `Id Element` element.
+
+### Members
+
+- (string)
+- (Id Element)
+
+## Id Element (Distinct Element)
+
+Element representing an identifier in the MSON namespace.
+
+### Properties
+
+- `element`: id (string, required, fixed)
+- `content` (string, required)
+
+## Distinct Element (Element)
```

Comment:
What is the `Element` here? Do you mean `Refract Element`? If not, where is the definition.